nomad/core_sched.go

```diff
@@ -286,7 +279,18 @@ func (c *CoreScheduler) gcEval(eval *structs.Evaluation, thresholdIndex uint64,
 	gcEval := true
 	var gcAllocIDs []string
 	for _, alloc := range allocs {
-		if !alloc.TerminalStatus() || alloc.ModifyIndex > thresholdIndex {
+		if job == nil || job.Stop || job.Status == structs.JobStatusDead {
```

Comment:
You still need to ensure that the alloc is terminal and modify index is less than threshold

Reply:
May want to move the logic into the gcEligible function

Reply:
When the job is stopped or dead wouldn't there be no allocs that are non terminal?

Reply:
Not necessarily. The scheduler still needs to run and mark the allocations as desired state stop.